src/viam/sdk/services/motion/client.cpp

```diff
@@ -45,13 +45,19 @@ std::string MotionClient::move_on_map(
     const Name& component_name,
     const Name& slam_name,
     const std::shared_ptr<motion_configuration>& motion_configuration,
+    const std::vector<GeometryConfig>& obstacles,
     const AttributeMap& extra) {
     return make_client_helper(this, *stub_, &StubType::MoveOnMap)
         .with(extra,
               [&](auto& request) {
                   *request.mutable_destination() = destination.to_proto();
                   *request.mutable_component_name() = component_name.to_proto();
                   *request.mutable_slam_service_name() = slam_name.to_proto();
+
+                  for (const auto& obstacle : obstacles) {
```

Comment:
So a user couldn't pass `nullptr` or anything like that because the arg is not a pointer. If they wanted to not specify any obstacles they would want to pass an empty `vec` (trivial to do, just pass `{}` as the arg). Nothing special is required in that case.